### ISSUE TITLE
[health 4.3.0] WorkoutType 17 causes crash

### ISSUE THREAD

**Issue by Ringert** [cph-cachet/flutter-plugins#619]
**Remember to specify the plugin name in the title!**

### Device / Emulator and OS
Please complete the following information for each phone and/or emulator you're experiencing this bug on:
 - Device: Samsung S22
 - OS: Android 12

**NB: Bugs pertaining to old devices/OS versions will likely not be fixed.**
* For iOS only the latest OS is supported. 
* For Android, see [the OS versions for which Google support security fixes](https://en.wikipedia.org/wiki/Android_version_history)**

### Describe the bug
WorkoutType 17 causes unhandled exceptions in HealthPlugin.kt:754.

**Exception:**
![image](https://user-images.githubusercontent.com/16204474/200834540-1e36a279-0c39-4c5c-b6bb-5c565067d5d7.png)

**Code:**
```
import 'dart:io';

import 'package:health/health.dart';
import 'package:permission_handler/permission_handler.dart';

class HealthSyncProvider {
  final HealthFactory _health = HealthFactory();

  HealthSyncProvider();

  Future<Iterable<HealthDataPoint>> _getNewHealthData() async {
    // define the types to get
    var types = [HealthDataType.WORKOUT];

    // requesting access to the data types before reading them
    await _health.requestAuthorization(types);

    var now = DateTime.now();

    // fetch health data
    List<HealthDataPoint> healthData = await _health.getHealthDataFromTypes(
        now.subtract(Duration(days: 24)), now, types);

    return healthData;
  }

  Future<bool> _requestAndroidPermissions() async {
    final activityPermission =
    await Permission.activityRecognition
        .request()
        .isGranted;
    final locationPermission = await Permission.location
        .request()
        .isGranted;

    return activityPermission && locationPermission;
  }

  Future<Iterable<HealthDataPoint>> getNewWorkoutData() async {
    if (Platform.isAndroid) {
      final permissionGranted = await this._requestAndroidPermissions();

      if (!permissionGranted) {
        return [];
      }
    }

    return (await this._getNewHealthData())
        .where((element) => element.value is WorkoutHealthValue);
  }
}
```

**Data from Google Fit Playground:**
```
{
  "session": [
    {
      "id": "3ab855309fa02e3f:watch-activemode:biking.spinning:1665985370319",
      "name": "Spinning am Morgen",
      "description": "",
      "startTimeMillis": "1665985370319",
      "endTimeMillis": "1665986614418",
      "modifiedTimeMillis": "1666735020206",
      "application": {
        "packageName": "com.google.android.apps.fitness"
      },
      "activityType": 17
    },
    {
      "id": "71b53161-4280-402e-8e3d-7a25fd492c7f",
      "name": "Rücken, Schultern, Trapez",
      "description": "",
      "startTimeMillis": "1665985879880",
      "endTimeMillis": "1665988274647",
      "modifiedTimeMillis": "1665991664632",
      "application": {
        "packageName": "com.anthonyng.workoutapp"
      },
      "activityType": 80
    },
    {
      "id": "3ab855309fa02e3f:watch-activemode:strength_training:1665986628525",
      "name": "Krafttraining am Morgen",
      "description": "",
      "startTimeMillis": "1665986628525",
      "endTimeMillis": "1665989078422",
      "modifiedTimeMillis": "1666735020206",
      "application": {
        "packageName": "com.google.android.apps.fitness"
      },
      "activityType": 80
    },
    {
      "id": "cb3115c5-a70f-4772-b0e4-75651f6da568",
      "name": "Improvisiertes Training",
      "description": "",
      "startTimeMillis": "1666330021328",
      "endTimeMillis": "1666331339808",
      "modifiedTimeMillis": "1666334923949",
      "application": {
        "packageName": "com.anthonyng.workoutapp"
      },
      "activityType": 80
    },
    {
      "id": "3ab855309fa02e3f:watch-activemode:strength_training:1666330227861",
      "name": "Krafttraining am Morgen",
      "description": "",
      "startTimeMillis": "1666330227861",
      "endTimeMillis": "1666332721134",
      "modifiedTimeMillis": "1666735020206",
      "application": {
        "packageName": "com.google.android.apps.fitness"
      },
      "activityType": 80
    }
  ],
  "deletedSession": []
}
```

### To Reproduce
Add a workout with type 17 (Spinning) and try to get it with the code above.

### Expected behavior
The app should not crash.

### Actual behavior
The app crashes.
![image](https://user-images.githubusercontent.com/16204474/200835042-0bee42b3-dcca-46f9-9408-ca5c7a137df7.png)

### Flutter doctor 
![image](https://user-images.githubusercontent.com/16204474/200835127-08a755d5-c128-4e9c-abe7-7d269280493a.png)

### Additional information
Add any other info about the problem here.

**Comment by irfangul92** [comment]
this is happening for some users in our app too. same crash looks like mapping is missing for some type of workouts

**Comment by sayedsaad07** [comment]
Any Updates on this issue. I can reproduce and its a big blocker for us

**Comment by marinosabijan** [comment]
Similar issue with me. Any fix?

**Comment by MadsVSChristensen** [comment]
Fixed in v4.5.0
